### PR TITLE
[TT-9734] Allow use of restricted API keys

### PR DESCRIPTION
**Pull request by chaddgrimm**
This PR adds a constructor parameter to the Geocoder class allowing users to pass in a referer, headers and host.

### CODE REVIEW

File: src/Geocoder.php [hunk 1]
```diff
@@ -136,26 +144,20 @@ public function isValid3wa($input)
...
   {
-    // add the apikey to the parameters
-    $parameters["key"] = $this->apiKey;
-
     // make an array out of the dictionary so that each element is a key value pair glued together with an '=', and urlencode the parameters
     $param_array = [];
     foreach ($parameters as $key => $value) {
       $param_array[] = "$key=" . urlencode($value);
     }
-    $param_array[] = "$key=" . urlencode($value);
```
Comment (imndaiga) [inline src/Geocoder.php:147]: 👀 good catch, not clear on the php runtime but strange this didn't break anything

File: src/Geocoder.php [hunk 2]
```diff
@@ -136,26 +144,20 @@ public function isValid3wa($input)
   // Prepare the call to the API server
   private function performRequest($command, $parameters)
   {
-    // add the apikey to the parameters
-    $parameters["key"] = $this->apiKey;
```
Comment (imndaiga) [inline src/Geocoder.php:140]: Might be worth documenting this as a deprecation - curious if switching to header-based authentication will improve our caching 🤔

**Review by imndaiga — APPROVED** [review]
LGTM!

File: src/Geocoder.php (hunk 2 above)
Comment (octogrimm) [reply src/Geocoder.php:140]: This is more of an internal detail on how we pass our api key to our api. I felt moving this to the header is a better approach (as to not expose the keys on the url).

File: src/Geocoder.php (hunk 1 above)
Comment (octogrimm) [reply src/Geocoder.php:147]: Yeah, bit of a weird one. Caught this while trying to debug the parameters passed to our API and saw there's an extra "key=null" at the end of the url. 😂

File: src/Geocoder.php (hunk 2 above)
Comment (chaddgrimm) [reply src/Geocoder.php:140]: This is more of an internal detail on how we pass our api key to our api. I felt moving this to the header is a better approach (as to not expose the keys on the url).

File: src/Geocoder.php (hunk 1 above)
Comment (chaddgrimm) [reply src/Geocoder.php:147]: Yeah this adds an extra `key=null` at the end of the params.

**Pull request merged**